netspeed/src/netspeed.c

```diff
@@ -1529,52 +1533,67 @@ update_tooltip (MateNetspeedApplet* applet)
     else {
         char ipv4_text [INET_ADDRSTRLEN];
         char ipv6_text [INET6_ADDRSTRLEN];
-        char *ip;
+        char *ip, *ipv6;
+
+        if (applet->devinfo->ip) {
+            format_ipv4 (applet->devinfo->ip, ipv4_text);
+            ip = ipv4_text;
+        } else {
+            ip = _("no IPv4");
+            ipv6 = _("no IPv6");
+        }
 
         if (applet->show_all_addresses) {
             format_ipv6 (applet->devinfo->ipv6, ipv6_text);
-            if (applet->devinfo->ip) {
-                format_ipv4 (applet->devinfo->ip, ipv4_text);
-                if (strlen (ipv6_text) > 2) {
-                    g_string_printf (tooltip,
-                                     _("%s: %s and %s"),
-                                     applet->devinfo->name,
-                                     ipv4_text,
-                                     ipv6_text);
-                } else {
-                    g_string_printf (tooltip,
-                                     _("%s: %s"),
-                                     applet->devinfo->name,
-                                     ipv4_text);
+
+            if (strlen (ipv6_text) > 2) {
+                struct ifaddrs *ifaddr;
+                int family, s;
+                ipv6 = "";
```

Comment:
You can initialize the variable to null and then check the value for printing the first line, which should not start with a line break.

Reply:
I agree about initialization. However, I don't see why it shouldn't start with a line break? In any case I would want the IPv6 addresses to start from the next line. But probably I'll rewrite the code to use `g_string_append_printf` and then loop through the suggested array of IPv6 addresses.

Reply:
see #568